plugins/outputs/kinesis/kinesis_test.go

```diff
@@ -578,11 +546,13 @@ func (m *mockKinesisPutRecords) AssertRequests(
 	expected []*kinesis.PutRecordsInput,
 ) {
 
-	assert.Equal(
+	if !assert.Equal(
 		len(expected),
 		len(m.requests),
 		fmt.Sprintf("Expected %v requests", len(expected)),
-	)
+	) {
+		return
```

Comment:
Exit early to avoid index out of range in subsequent check

Reply:
You can achieve the same using `require.Equal`. Citing the [package documentation](https://pkg.go.dev/github.com/stretchr/testify/require):

> Package require implements the same assertions as the `assert` package but stops test execution when a test fails.

Reply:
So pretty much avoid `testify/assert` and use `testify/require`.  :)